src/main/webapp/app/course/competencies/components/course-competency-relation-form/course-competency-relation-form.component.ts

```
@@ -0,0 +1,289 @@
+    protected resetForm() {
+        this.headCompetencyId.set(undefined);
+        this.tailCompetencyId.set(undefined);
+        this.relationType.set(undefined);
+        this.onRelationDeselection.emit();
+    }
+
+    protected async createRelation(): Promise<void> {
+        try {
+            this.isLoading.set(true);
+            const courseCompetencyRelation = await this.courseCompetencyApiService.createCourseCompetencyRelation(this.courseId(), {
+                headCompetencyId: this.headCompetencyId()!,
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)